packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -418,8 +419,23 @@ export async function getAvailableSlots({ input, ctx }: GetScheduleOptions): Pro
       input.eventTypeId
     }`
   );
+
+  // The effectiveMinimumBookingNotice just makes sure we don't discard slots too early, so we can later decide
+  // which minimumBookingNotice to apply based on if the slot has any attendees or not.
+  let seatsMinimumBookingNoticeActive = false;
+  let effectiveMinimumBookingNotice = eventType.minimumBookingNotice;
+  // not using ternary's here to help TS narrow down the type of effectiveMinimumBookingNotice correctly
+  if (
+    eventType.seatsPerTimeSlot &&
+    eventType.seatsMinimumBookingNotice !== null &&
+    eventType.seatsMinimumBookingNotice < eventType.minimumBookingNotice
+  ) {
+    seatsMinimumBookingNoticeActive = true;
+    effectiveMinimumBookingNotice = eventType.seatsMinimumBookingNotice;
```

Comment:
Why not simplify and go for something like this? Untested pseudocode.

```javascript
const effectiveMinimumBookingNotice = !isNaN(seatsMinimumBookingNotice) && seatsCount > 0 
  ? Math.min(seatsMinimumBookingNotice, minimumBookingNotice) 
  : minimumBookingNotice;
```

@CarinaWolli and I also had a discussion just now that may be simplifying further also; we see no reason to continue applying a minimum booking notice once a booking has been booked for further seats. So the seatsMinimumBookingNotice is possibly default-able to `0`

Reply:
> So the seatsMinimumBookingNotice is possibly default-able to 0

Also in the input field we should default to 0 for `seatsMinimumBookingNotice`, so it's clear to the user